src/org/pentaho/agilebi/modeler/models/annotations/AnnotationType.java

```diff
@@ -359,6 +375,169 @@ public void populate( final Map<String, Serializable> propertiesMap ) {
     }
   }
 
+  /**
+   * Returns the physical column name that this annotation should operate on. For sources
+   * based on HierarchyLevels we need to consult the existing model to find the underlying
+   * source field.
+   *
+   * @param modelerWorkspace Workspace to look for the level
+   * @param levelName Level formula to find the level
+   * @param cubeName Cube name to file the level
+   * @return field
+   * @throws ModelerException
+   */
+  protected String resolveFieldFromLevel( final ModelerWorkspace modelerWorkspace,
+                                          final String levelName,
+                                          final String cubeName ) throws ModelerException {
+    if ( !StringUtils.isBlank( levelName ) && !StringUtils.isBlank( cubeName ) ) {
+      String locale = Locale.getDefault().toString();
+      LogicalModel businessModel = modelerWorkspace.getLogicalModel( ModelerPerspective.ANALYSIS );
+      List<OlapCube> olapCubes = (List<OlapCube>) businessModel.getProperty( OLAP_CUBES_PROPERTY );
```

Comment:
Since this logic for getting the businessModel and cubes are the same between this and the resolveFieldFromMeasure, it could be abstracted to its own helper method